### PR TITLE
ENH: Image sequence in zipfiles

### PR DESCRIPTION
**Pull request by caspervdw**
When a path to a zipfile is specified, all files inside the zipfile will be loaded as an imagesequence using `PIL.Image.open`. For large collections of images, it may be preferred to work from zipfiles.

### CODE REVIEW

File: pims/image_sequence.py [hunk 1]
```diff
@@ -16,6 +19,7 @@
 
 # skimage.io.plugin_order() gives a nice hierarchy of implementations of imread.
 # If skimage is not available, go down our own hard-coded hierarchy.
+from PIL import Image
```
Comment (tacaswell) [inline pims/image_sequence.py]: I always get confused with the PIL/pillow conventions.  Does this work with both?  We should prefer pillow to PIL.

File: pims/image_sequence.py [hunk 2]
```diff
@@ -102,6 +121,17 @@ def _get_files(self, path_spec):
             self._count = len(path_spec)
             return
 
+        if zipfile.is_zipfile(path_spec):
+            self.is_zipfile = True
```
Comment (tacaswell) [inline pims/image_sequence.py]: For readability reasons can you put a `self._is_zipfile = False` and `self._zipfile = None` in the `__init__`?  I also suggest making these 'private' attributes as these details should be mostly hidden from the users.

Comment (tacaswell) [inline pims/image_sequence.py]: Can you put either an exception or a warning in here if the user passes both a zipfile _and_ a plugin?  Silently ignoring user input is a bad thing.

**Comment by tacaswell** [comment]
Awesome!

I left a few style comments, but other wise looks good to me.

File: pims/image_sequence.py (hunk 1 above)
Comment (caspervdw) [inline pims/image_sequence.py]: I tested it with Pillow only, but I think `Image.open` has the same syntax in both versions.

Comment (tacaswell) [inline pims/image_sequence.py]: Great, then disregard this comment.

**Comment by caspervdw** [comment]
I implemented the changes you suggested and additionally made it compatible with Python 3.x . I think it can be merged.

File: pims/image_sequence.py [hunk 3]
```diff
@@ -94,6 +101,20 @@ def __init__(self, path_spec, process_func=None, dtype=None,
         else:
             self._dtype = dtype
 
+    def close(self):
+        if self._is_zipfile:
+            self._zipfile.close()
+
+    def __del__(self):
```
Comment (tacaswell) [inline pims/image_sequence.py]: Can you push this up to the base-class (along with the context manager stuff)?

File: pims/image_sequence.py [hunk 4]
```diff
@@ -94,6 +101,20 @@ def __init__(self, path_spec, process_func=None, dtype=None,
         else:
             self._dtype = dtype
 
+    def close(self):
```
Comment (tacaswell) [inline pims/image_sequence.py]: Probably worth using a `super` call in here to get the base-class close.

**Comment by tacaswell** [comment]
Sorry, two more minor things, but looks great other wise!

File: pims/image_sequence.py (hunk 3 above)
Comment (caspervdw) [inline pims/image_sequence.py]: I am not sure if it's a good idea to define the destructor to call `close` in all cases. If you call `frames.close()`, and then destroy the object, `close()` will be run twice which might give trouble for some readers.

For the zipfile it's no problem to close it twice, that's why I put it in for this specific reader.

`__enter__` `__exit__` already works by defining `close()`, right?

**Comment by tacaswell** [comment]
Thanks!

**Pull request merged**